parsl/executors/workqueue/executor.py

```diff
@@ -808,6 +411,55 @@ def _serialize_function(self, fn_path, parsl_fn, parsl_fn_args, parsl_fn_kwargs)
         with open(fn_path, "wb") as f_out:
             pickle.dump(function_info, f_out)
 
+    def _construct_map_file(self, map_file, input_files, output_files):
+        """ Map local filepath of parsl files to the filenames at the execution worker.
+        If using a shared filesystem, the filepath is mapped to its absolute filename.
+        Otherwise, to its original relative filename. In this later case, work queue
+        recreates any directory hierarchy needed."""
+        file_translation_map = {}
+        for spec in itertools.chain(input_files, output_files):
+            local_name = spec[0]
+            if self.shared_fs:
+                remote_name = os.path.abspath(local_name)
+            else:
+                remote_name = local_name
+            file_translation_map[local_name] = remote_name
+        with open(map_file, "wb") as f_out:
+            pickle.dump(file_translation_map, f_out)
+
+    def _register_file(self, parsl_file):
+        """Generates a tuple (parsl_file.filepath, stage, cache) to give to
+        work queue. cache is always False if self.use_cache is False.
+        Otherwise, it is set to True if parsl_file is used more than once.
+        stage is True if the file needs to be copied by work queue. (i.e., not
+        a URL or an absolute path)
+
+        It has the side-effect of adding parsl_file to a list of registered
+        files.
+
+        Note: The first time a file is used cache is set to False. Since
+        tasks are generated dynamically, without other information this is
+        the best we can do."""
+        to_cache = False
+        if self.use_cache:
+            to_cache = parsl_file in self.registered_files
+
+        to_stage = False
+        if parsl_file.scheme == 'file' or (parsl_file.local_path and os.path.exists(parsl_file.local_path)):
+            to_stage = not os.path.isabs(parsl_file.filepath)
+
+        self.registered_files.add(parsl_file)
+
+        return ParslFileToWq(parsl_file.filepath, to_stage, to_cache)
+
+    def _std_output_to_wq(self, fdname, stdfspec):
+        """Find the name of the file that will contain stdout or stderr and
+        return a ParslFileToWq with it. These files are never cached"""
+        fname, mode = putils.get_std_fname_mode(fdname, stdfspec)
+        to_stage = not os.path.isabs(fname)
+        print(ParslFileToWq(fname, stage=to_stage, cache=False))
```

Comment:
use logger or remove?

Reply:
Removed in 2a6830c04ffebf34fb7170f9a1857eebc7e4fc62